go/vt/worker/split_diff.go

```diff
@@ -515,7 +551,18 @@ func (sdw *SplitDiffWorker) diff(ctx context.Context) error {
 			if key.KeyRangeEqual(overlap, sdw.shardInfo.KeyRange) {
 				destinationQueryResultReader, err = TableScan(ctx, sdw.wr.Logger(), sdw.wr.TopoServer(), sdw.destinationAlias, tableDefinition)
```

Comment:
This doesn't have the comment above but I want to make sure I understand the flow: We also expect this to _never_ get hit in a merge case, right?